corehq/apps/reports/standard/cases/case_list_explorer.py

```diff
@@ -99,6 +99,18 @@ def _populate_sort(self, query, sort):
             column = self.headers.header[column_id]
             try:
                 special_property = SPECIAL_CASE_PROPERTIES_MAP[column.prop_name]
+                if special_property.key == '@case_id':
+                    # This condition is added because ES 5 does not allow sorting on _id.
+                    #  When we will have case_id in root of the document, this should be removed.
+                    sort_order = 'desc' if descending else 'asc'
+                    query.es_query['sort'] = [{
+                        'case_properties.value.exact': {
+                            'order': sort_order,
+                            'nested_path': 'case_properties',
+                            'nested_filter': {'term': {"case_properties.key.exact": "@case_id"}},
+                        }
+                    }]
+                    return query
```

Comment:
```suggestion
                    query = query.sort_by_case_property('value.exact', desc=descending)
```
I don't recall why I didn't do this with the case API sorting - I thought it was because it sorts by multiple fields, but I see `nested_sort` supports `reset_sort=False`, so presumably that would've worked.

Reply:
This is so much better. I will update the code with the suggestion.